src/Integration.Vsix/CFamily/VcxProject/FileConfig.cs

```diff
@@ -28,8 +29,99 @@ namespace SonarLint.VisualStudio.Integration.Vsix.CFamily.VcxProject
+            {
+                logger.WriteLine("\"ClCompilerPath\" was not found.");
+                return false;
+            }
+
+            if (!fileSystem.File.Exists(compilerPath))
+            {
+                logger.WriteLine($"Compiler path \"{compilerPath}\" does not exist.");
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool TryGetCompilerPathFromExecutablePath(ILogger logger, VCConfiguration vcConfig, IFileSystem fileSystem, out string compilerPath)
+        {
+            compilerPath = default;
+            var executablePath = vcConfig.GetEvaluatedPropertyValue("ExecutablePath");
+            if (string.IsNullOrEmpty(executablePath))
+            {
+                logger.WriteLine("\"ExecutablePath\" was not found.");
+                return false;
+            }
+
+            var toolExe = vcConfig.GetEvaluatedPropertyValue("CLToolExe");
+            if (string.IsNullOrEmpty(toolExe))
+            {
+                logger.WriteLine("\"CLToolExe\" was not found.");
+                return false;
+            }
+
+            foreach (var path in executablePath.Split(';'))
+            {
+                compilerPath = Path.Combine(path, toolExe);
+                if (fileSystem.File.Exists(compilerPath))
+                {
+                    return true;
+                }
+                else
+                {
+                    logger.WriteLine($"Compiler path \"{compilerPath}\" does not exist.");
+                }
+            }
+
+            return false;
+        }
+
+        private static bool TryGetCompilerPathFromVCExecutablePath(ILogger logger, VCConfiguration vcConfig, IFileSystem fileSystem, out string compilerPath)
+        {
+            var platform = ((VCPlatform)vcConfig.Platform).Name.Contains("64") ? "x64" : "x86";
+            var exeVar = "VC_ExecutablePath_" + platform;
+            compilerPath = Path.Combine(vcConfig.GetEvaluatedPropertyValue(exeVar), "cl.exe");
+            if (fileSystem.File.Exists(compilerPath))
+            {
+                return true;
+            }
+            else
+            {
+                logger.WriteLine($"Compiler path \"{compilerPath}\" does not exist.");
+                return false;
+            }
+        }
+
+        private static bool GetCompilerPath(ILogger logger, VCConfiguration vcConfig, IFileSystem fileSystem, out string compilerPath)
+        {
+            if (TryGetCompilerPathFromClCompilerPath(logger, vcConfig, fileSystem, out compilerPath))
+            {
+                return true;
+            }
+
+            // Fallback to ExecutablePath and CLToolExe
+            if (TryGetCompilerPathFromExecutablePath(logger, vcConfig, fileSystem, out compilerPath))
+            {
+                return true;
+            }
+
+            // Fallback to VC_ExecutablePath, which is used to be used in VS2017 toolchains
+            // because ClCompilerPath was not available
```

Comment:
```suggestion
            // In case ClCompilerPath isn't available, and ExecutablePath matching fails.
```
Feel free to adjust my suggestion. The old comment doesn't mention anything about ExecutablePath...